tools/api/clone.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+
+# Simple script to clone vM to vN API, performing sed-style heuristic fixup of
+# build paths and package references.
+#
+# Usage:
+#
+# ./tools/api/clone.sh v2 v3
+
+set -e
+
+declare -r OLD_VERSION="$1"
+declare -r NEW_VERSION="$2"
+
+function replace_build() {
```

Comment:
For sed weaklings, can you provide a brief comment/example above each function of this thing does? Same below.